pallets/xdns/src/lib.rs

```diff
@@ -769,6 +754,51 @@ pub mod pallet {
                 allowed_side_effects,
             )
         }
+
+        pub fn update_epoch_history(
+            verifier: &GatewayVendor,
+            epoch_duration_in_remote_blocks: T::BlockNumber,
+            epoch_duration_in_local_blocks: T::BlockNumber,
+        ) {
+            const N: u32 = 10; // Number of epochs to consider for the moving average
+
+            let mut history = EpochHistory::<T>::get(verifier).unwrap_or_default();
+            let history_len = history.len() as u32;
+
+            if history_len >= N {
```

Comment:
How about the case where `history_len` is (way) larger than `N`? By making `history.remove(0)`, is it enough? Or should be something like `history.remove(history.len() as u32 - N)`?

Reply:
that's a good comment actually 👍